### PR TITLE
Gatekeeper init gsd

### PR DESCRIPTION
**Pull request by borditamas**

### CODE REVIEW

File: gatekeeper/src/main/java/eu/arrowhead/core/gatekeeper/service/GatekeeperDriver.java [hunk 1]
```diff
@@ -134,6 +134,7 @@ public void onApplicationEvent(final ContextRefreshedEvent event) {
 				
 			} catch (final InterruptedException ex) {
 				logger.trace("Thread {} is interrupted...", Thread.currentThread().getName());
+				gsdPollRequestExecutor.shutdownExecutionNow();
```
Comment (rbocsi) [inline gatekeeper/src/main/java/eu/arrowhead/core/gatekeeper/service/GatekeeperDriver.java:137]: rethrow the exception after shutdown executor

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollRequestExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		doThrow(RejectedExecutionException.class).when(threadPool).execute(any());
+		
+		testingObject.execute();
+		
+		final GSDPollResponseDTO gsdPollResponseDTO = queue.take();
+		
+		assertNull(gsdPollResponseDTO.getProviderCloud());
+	}
+	
+	//=================================================================================================
```
Comment (rbocsi) [inline gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollRequestExecutorTest.java:66]: unused method

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+import eu.arrowhead.core.gatekeeper.relay.GatekeeperRelayResponse;
+import eu.arrowhead.core.gatekeeper.relay.GeneralAdvertisementResult;
+
+@RunWith(SpringRunner.class)
+public class GSDPollTaskTest {
+	
+	//=================================================================================================
+	// members
+	
+	@InjectMocks
```
Comment (rbocsi) [inline gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java]: remove this annotation

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+		final GSDPollResponseDTO gsdPollResponseDTO = queue.take();
+		
+		assertNull(gsdPollResponseDTO.getProviderCloud());
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Test
+	public void testRunWithThrowingJMSExceptionByRelayClient1() throws JMSException, InterruptedException {
+		doThrow(JMSException.class).when(relayClient).publishGeneralAdvertisement(any(), any(), any());
+		when(relayClient.sendRequestAndReturnResponse(any(), any(), any())).thenReturn(new GatekeeperRelayResponse("session-id", "message-type", new GSDPollResponseDTO()));
```
Comment (rbocsi) [inline gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java]: is this line necessary?

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GatekeeperDriverGSDTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import eu.arrowhead.common.dto.ServiceQueryFormDTO;
+import eu.arrowhead.common.dto.ServiceRegistryResponseDTO;
+
+@RunWith(SpringRunner.class)
+public class GatekeeperDriverGSDTest {
+	
+	//=================================================================================================
+	// members
+		
+	@InjectMocks
```
Comment (rbocsi) [inline gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GatekeeperDriverGSDTest.java]: there is no mocks in the class

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollRequestExecutorTest.java (hunk 2 above)
Comment (borditamas) [reply gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollRequestExecutorTest.java:66]: It's needed as discussed

File: gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java (hunk 4 above)
Comment (borditamas) [reply gatekeeper/src/test/java/eu/arrowhead/core/gatekeeper/service/GSDPollTaskTest.java]: actually no...

**Review by rbocsi — APPROVED** [review]

**Pull request merged**